app/Controllers/LanguagePackageController.js

```diff
@@ -22,26 +27,55 @@ async function addLanguagePackage(req, res) {
 
 async function sendLanguagePackages(req, res) {
   // get userId from request
-  const { id } = await req.user;
+  const { id } = req.user;
+  const { groups } = req.query || false;
 
   // get language Package
   const languagePackages = await getLanguagePackages(id, res);
```

Comment:
could you add a parameter groups to this function to decide if we wan't to include the groups insted of this messy if solution.

```diff
diff --git a/app/Controllers/LanguagePackageController.js b/app/Controllers/LanguagePackageController.js
index 3a54942..6189a50 100644
--- a/app/Controllers/LanguagePackageController.js
+++ b/app/Controllers/LanguagePackageController.js
@@ -5,7 +5,6 @@ const {
 } = require('../Services/LanguagePackageServiceProvider.js');
 const { createDrawers } = require('../Services/DrawerServiceProvider.js');
 const { drawers } = require('../utils/constants.js');
-const { getGroups } = require('../Services/GroupServiceProvider.js');
 const {
   getNumberOfUnresolvedVocabulary,
   getNumberOfUnactivatedVocabulary,
@@ -31,35 +30,18 @@ async function sendLanguagePackages(req, res) {
   const { groups } = req.query || false;

   // get language Package
-  const languagePackages = await getLanguagePackages(id, res);
+  const languagePackages = await getLanguagePackages(id, res, groups);

-  let formatted;
-  // if groups is true, return groups to every language package
-  if (groups) {
-    formatted = await Promise.all(
-      languagePackages.map(async (languagePackage) => ({
-        unresolvedVocabularies: await getNumberOfUnresolvedVocabulary(languagePackage.id, id),
+  const formatted = await Promise.all(
+    languagePackages.map(async (languagePackage) => ({
+      unresolvedVocabularies: await getNumberOfUnresolvedVocabulary(languagePackage.id, id),

-        // add number of unactivated vocabularies
-        unactivatedVocabularies: await getNumberOfUnactivatedVocabulary(languagePackage.id, id),
+      // add number of unactivated vocabularies
+      unactivatedVocabularies: await getNumberOfUnactivatedVocabulary(languagePackage.id, id),

-        groups: await getGroups(id, languagePackage.id, res),
-
-        ...languagePackage.toJSON(),
-      }))
-    );
-  } else {
-    formatted = await Promise.all(
-      languagePackages.map(async (languagePackage) => ({
-        unresolvedVocabularies: await getNumberOfUnresolvedVocabulary(languagePackage.id, id),
-
-        // add number of unactivated vocabularies
-        unactivatedVocabularies: await getNumberOfUnactivatedVocabulary(languagePackage.id, id),
-
-        ...languagePackage.toJSON(),
-      }))
-    );
-  }
+      ...languagePackage.toJSON(),
+    }))
+  );

   res.send(formatted);
 }
diff --git a/app/Services/LanguagePackageServiceProvider.js b/app/Services/LanguagePackageServiceProvider.js
index 6ce7316..d529e3e 100644
--- a/app/Services/LanguagePackageServiceProvider.js
+++ b/app/Services/LanguagePackageServiceProvider.js
@@ -1,4 +1,4 @@
-const { LanguagePackage } = require('../../database');
+const { LanguagePackage, Group } = require('../../database');

 // create language package
 async function createLanguagePackage(
@@ -18,9 +18,10 @@ async function createLanguagePackage(
 }

 // get language package
-async function getLanguagePackages(userId, res) {
+async function getLanguagePackages(userId, res, groups) {
   // Get user with email from database
   const languagePackages = await LanguagePackage.findAll({
+    include: groups ? [{ model: Group, attributes: ['id', 'name', 'active'] }] : [],
     attributes: ['id', 'name', 'foreignWordLanguage', 'translatedWordLanguage', 'vocabsPerDay', 'rightWords'],
     where: {
       userId: userId,
```